### PR TITLE
Demo Parsing Debugging

### PR DESCRIPTION
**Pull request by ewega**
Fixes Demo Parsing Errors by catching the following:

1) Discards any event happening before a given round (therefore includes events between rounds too)

2) Discards any event happening before a round END if it doesn't have a round START

3) Discards duplicate round STARTS

**Test on multiple demo files**

### CODE REVIEW

File: demofile-parsing/parse_demo.js [hunk 1]
```diff
@@ -3,71 +3,80 @@ const demofile = require("demofile");
 const path = require("path");
 const Round = require("./Round.js");
 
-const demPath = "vitality-vs-nip-dust2.dem"; //add the filepath here
-const gameStart = 13301;
```
Comment (mattlianje) [inline demofile-parsing/parse_demo.js:7]: yah - I think Lam is still werking on this. @tjlam thots? Seems fine to me though

File: demofile-parsing/parse_demo.js [hunk 2]
```diff
@@ -2,72 +2,82 @@ const fs = require("fs");
...
   const demoFile = new demofile.DemoFile();
   const rounds = [];
+  var roundEvents = [];
+  var roundLatestStart = 0;
+  var roundOn = false;
+  var roundIndex = 0;
+
   // parse start of the round
   demoFile.gameEvents.on("round_start", e => {
+    roundOn = true;
```
Comment (mattlianje) [inline demofile-parsing/parse_demo.js:17]: Round on 'if wrappers' are covering everything. Works like a charm

**Review by mattlianje — APPROVED** [review]
image
Works as expected. The only nit would be the abundance of commented out logs but they are still useful at this point.

File: demofile-parsing/parse_demo.js (hunk 1 above)
Comment (tjlam) [reply demofile-parsing/parse_demo.js:7]: Yeah delete that boi if you need

File: demofile-parsing/parse_demo.js [hunk 3]
```diff
@@ -2,72 +2,82 @@ const fs = require("fs");
...
-    // r.getHighRateTimes();
-    // r.mergeTimeRanges();
-    // r.mapToStreamTimestamps(gameStart);
-    // r.plotAllEvents();
-    // r.plotHighRates();
-    console.log(`round ${roundIndex} ended`);
-    roundIndex++;
-  });
+    if(roundOn == true){
+      roundOn == false;
```
Comment (tjlam) [inline demofile-parsing/parse_demo.js:27]: I think this should be `roundOn = false`

**Pull request merged**